pykechain/models/base_reference.py

```diff
@@ -0,0 +1,129 @@
+from abc import abstractmethod, ABC
+from typing import Optional, Any, Text, Iterable, Union, Tuple, List
+
+from pykechain.models import Property2, Base
+from pykechain.models.base import BaseInScope
+from pykechain.utils import is_uuid
+
+
+class _ReferenceProperty(Property2):
+    """
+    Private base class for the KE-chain reference properties.
+
+    .. versionadded:: 3.7
+    :cvar REFERENCED_CLASS: handle to the Pykechain class that is referenced using this ReferenceProperty class.
+    """
+
+    REFERENCED_CLASS = Base  # type: type(Base)
```

Comment:
smart to create a 'base class' for the Referenced Properties. That is also how we do it in KEC backend.